ansible/roles/artifactory/tasks/main.yml

```diff
@@ -1,4 +1,46 @@
 ---
+- name: Install LVM Tools
+  ansible.builtin.dnf:
+    name: lvm2
+    state: latest
+
+- name: Create {{ artifactory_var_vg_name }} volume group
+  community.general.lvg:
+    vg: "{{ artifactory_var_vg_name }}"
+    state: present
+    pvs:
+      - "{{ artifactory_var_device_node }}"
+
+- name: Create {{ artifactory_var_lv_name }} logical volume
+  community.general.lvol:
+    lv: "{{ artifactory_var_lv_name }}"
+    vg: "{{ artifactory_var_vg_name }}"
+    resizefs: true
+    size: +100%FREE
+    state: present
+
+- name: Create xfs filesystem on the logical volume
+  community.general.filesystem:
+    dev: "/dev/{{ artifactory_var_vg_name }}/{{ artifactory_var_lv_name }}"
+    fstype: xfs
+    state: present
+
+- name: Create Artifactory var directory mount point
+  ansible.builtin.file:
+    path: "{{ artifactory_var_directory_path }}"
+    owner: root
+    group: root
+    mode: '0755'
+    state: directory
+
+- name: Mount Artifactory var directory
+  ansible.posix.mount:
+    path: "{{ artifactory_var_directory_path }}"
+    src: "/dev/{{ artifactory_var_vg_name }}/{{ artifactory_var_lv_name }}"
+    fstype: xfs
+    opts: defaults,noatime
+    state: mounted
+
```

Comment:
Could this block of tasks be replaced with the [ansible-role-mounted-volume](https://github.com/companieshouse/ansible-role-mounted-volume) role and some host configuration? The role doesn't currently support filesystem options but that would be a small change that could be introduced in a new version.

Reply:
Updated in latest commits